metricbeat/module/elasticsearch/elasticsearch.go

```diff
@@ -21,17 +21,54 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/url"
+	"sync"
+	"time"
 
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/metricbeat/helper"
 )
 
 // Global clusterIdCache. Assumption is that the same node id never can belong to a different cluster id
 var clusterIDCache = map[string]string{}
 
+// Global cache for license information. Assumption is that license information changes infrequently
+type _licenseCache struct {
```

Comment:
Why do you start with a _ here? I would prefer not to have _ in var names.

Reply:
I don't like the `_` prefix here either so I'm open to other suggestions. The reason I did this was because, just a few lines below, I want to create the actual singleton cache instance variable. I wanted _that_ variable to have the "real" name, `licenseCache`, so I had to come up with some other name for the type.